urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/examples/PairsInAv231.java

```diff
@@ -0,0 +1,243 @@
+        }
+        System.out.println(Arrays.toString(cCounts));
+        c = new PermutationClass("231", "321");
+        for (Permutation q : new Permutations(c, 4)) {
+            PermutationClass d = new PermutationClass("231", "321", q.toString());
+            for (int n = 4; n < 12; n++) {
+                PermCounter counter = new PermCounter();
+                d.processPerms(n, counter);
+                cCounts[n - 4] = counter.getCount();
+            }
+            System.out.println(q + " " + Arrays.toString(cCounts));
+        }
+    }
+
+    public static void doWilf(int n, boolean verbose, int checkTotal) {
+        PermutationClass c = new PermutationClass("231");
+        HashSet<PermPair> pairs = new HashSet<>();
+        for (Permutation a : new Permutations(c, n)) {
+            for (Permutation b : new Permutations(c, n)) {
+                if (!a.equals(b)) {
+                    pairs.add(new PermPair(a, b));
+                }
+            }
+        }
+        HashSet<PermPair> pairReps = new HashSet<>();
+        for (PermPair pair : pairs) {
+            PermPair np = new PermPair(Symmetry.IRC.on(pair.a), Symmetry.IRC.on(pair.b));
+            if (!pairReps.contains(np) && !pairReps.contains(pair)) {
+                pairReps.add(pair);
+            }
+        }
+        System.out.println("Length " + n + " has " + pairReps.size() + " symmetry orbits.");
+        HashMap<Spectrum, HashSet<PermPair>> wilfClasses = new HashMap<>();
+        for (PermPair pair : pairs) {
+            long[] qC = new long[checkTotal];
+            PermutationClass d = new PermutationClass(new Permutation("231"), pair.a, pair.b);
+            for (int m = n + 1; m <= n + checkTotal; m++) {
+                PermCounter counter = new PermCounter();
+                d.processPerms(m, counter);
+                qC[m - n - 1] = counter.getCount();
+            }
+            // System.out.println(q + " " + Arrays.toString(qC));
+            Spectrum qs = new Spectrum(qC);
+            if (wilfClasses.containsKey(qs)) {
+                wilfClasses.get(qs).add(pair);
+            } else {
+                HashSet<PermPair> aSet = new HashSet<>();
+                aSet.add(pair);
+                wilfClasses.put(qs, aSet);
+            }
+        }
+        System.out.println("Length " + n + " has " + wilfClasses.keySet().size() + " classes");
+        if (verbose) {
+            System.out.println();
+            for (Spectrum s : wilfClasses.keySet()) {
+                System.out.println(Arrays.toString(s.counts) + " " + (wilfClasses.get(s).size()));
+                for (PermPair q : wilfClasses.get(s)) {
+                    System.out.println(q.a + " " + q.b);
+                    // System.out.println();
+                }
+                System.out.println();
+            }
+        }
+        wilfClasses.clear();
+    }
+
+    public static void doWilf(int n, int m, boolean verbose, int checkTotal) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 111-118, lines 119-126 into separate methods.